spec/models/workday/submit_customer_reversal_invoice_adjustment_request_spec.rb

```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+RSpec.describe Workday::SubmitCustomerReversalInvoiceAdjustmentRequest do
+  let(:user) { FactoryBot.create(:user, name: 'Forename Surname') }
+  let(:submission) do
+    FactoryBot.create(:submission_with_validated_entries,
+                      purchase_order_number: '123',
+                      submitted_by: user,
+                      task: task)
+  end
+  let(:framework) { submission.framework }
+  let(:task) { FactoryBot.create(:task, period_month: 12, period_year: 2018) }
+  let(:supplier) { submission.supplier }
+  let(:request) { Workday::SubmitCustomerReversalInvoiceAdjustmentRequest.new(submission) }
+
+  before do
+    commercial_agreements = double(
+      revenue_category_ids: { framework.short_name => 'Revenue_Category_WID' },
+      tax_code_ids: { framework.short_name => 'GBC20' }
+    )
+    allow(Workday::CommercialAgreements).to receive(:new).and_return(commercial_agreements)
+  end
+
+  it_behaves_like 'a workday request'
+
+  describe '#content' do
+    it_behaves_like 'an authenticated workday request'
+
+    it 'sets Company_Reference with the CCS Workday Company_Reference_ID' do
+      expect(
+        text_at_xpath("//ns0:Company_Reference//ns0:ID[@ns0:type='Company_Reference_ID']")
+      ).to eq 'Crown_Commercial_Service'
+    end
+
+    it 'sets Customer_Reference with the Supplier’s salesforce_id identified as Customer_Reference_ID' do
+      expect(
+        text_at_xpath("//ns0:Customer_Reference//ns0:ID[@ns0:type='Customer_Reference_ID']")
+      ).to eq supplier.salesforce_id
+    end
+
+    it 'sets From_Date with the date the Submission is being made against' do
+      expect(text_at_xpath('//ns0:From_Date')).to eq task.period_date.to_s
+    end
+
+    it 'sets Customer_PO_Number with the Submission’s purchase order number' do
+      expect(text_at_xpath('//ns0:Customer_PO_Number')).to eq submission.purchase_order_number
+    end
+
+    it 'sets Memo with the Submission’s ID' do
+      expect(text_at_xpath('//ns0:Memo')).to eq "Submission ID: #{submission.id}"
+    end
+
+    # Question: the original submitter, or the user submitting the correction that triggers this reversal?
+    it 'sets Note_Data with the name of the user who submitted the Submission' do
+      expect(text_at_xpath('//ns0:Note_Data//ns0:Note_Content')).to eq 'Forename Surname'
+    end
+
+    it 'sets the invoice as submitted' do
```

Comment:
it looks like this matches the specs for `SubmitCustomerInvoiceAdjustmentRequest`, but I would have expected this to read `'sets the invoice **adjustment** as submitted``